server/protocols/jmap-draft/src/main/java/org/apache/james/jmap/draft/methods/SetMailboxesUpdateProcessor.java

```diff
@@ -266,14 +269,20 @@ private void updateMailbox(Mailbox mailbox, MailboxUpdateRequest updateRequest,
                     .toMailboxAcl(),
                 mailboxSession);
         }
-        if (!originMailboxPath.equals(destinationMailboxPath)) {
-            mailboxManager.renameMailbox(mailbox.getId(), destinationMailboxPath, mailboxSession);
 
-            subscriptionManager.unsubscribe(mailboxSession, originMailboxPath.getName());
-            subscriptionManager.subscribe(mailboxSession, destinationMailboxPath.getName());
+        if (!originMailboxPath.equals(destinationMailboxPath)) {
+            List<MailboxRenamedResult> mailboxRenamedResults = mailboxManager.renameMailbox(mailbox.getId(), destinationMailboxPath, mailboxSession);
+            updateSubcription(mailboxRenamedResults, mailboxSession);
         }
     }
 
+    private void updateSubcription(List<MailboxRenamedResult> mailboxRenamedResults, MailboxSession mailboxSession) {
+        mailboxRenamedResults.forEach(Throwing.consumer(mailboxRenamedResult -> {
+            subscriptionManager.unsubscribe(mailboxSession, mailboxRenamedResult.getOriginPath().getName());
```

Comment:
you can unsubscribe a path that doesn't exist?

Reply:
Yes